domain/src/main/java/me/androidbox/domain/authentication/usecase/imp/DeleteCurrentUserUseCaseImp.kt

```diff
@@ -0,0 +1,14 @@
+package me.androidbox.domain.authentication.usecase.imp
+
+import me.androidbox.domain.authentication.preference.PreferenceRepository
+import me.androidbox.domain.authentication.usecase.DeleteCurrentUserUseCase
+import javax.inject.Inject
+
+class DeleteCurrentUserUseCaseImp @Inject constructor(
+    private val preferenceRepository: PreferenceRepository
+) : DeleteCurrentUserUseCase {
```

Comment:
There are zero reasons to abstract a use case. A use case is intended to contain business logic which you always want to test. There's no case in which you'd want to replace your real app's business logic with fake logic. 

General rule: If you only need a single implementation of something, don't abstract it. Just leads to tons of boilerplate code